### PR TITLE
github-commenter: add darwin platform

### PR DESCRIPTION
**Pull request by onsails**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/github-commenter/default.nix [hunk 1]
```diff
@@ -20,6 +20,6 @@ buildGoPackage rec {
     license = licenses.asl20;
     homepage = "https://github.com/cloudposse/github-commenter";
     maintainers = [ maintainers.mmahut ];
-    platforms = platforms.linux;
+    platforms = with platforms; linux ++ darwin;
```
Comment (prusnak) [inline pkgs/development/tools/github-commenter/default.nix:23]:
```suggestion
    platforms = platforms.unix;
```

**Review by prusnak — CHANGES_REQUESTED** [review]
Latest is 0.8.0. Do you want to try to update the package?

The new version uses Go modules, so you should use buildGoModule instead of buildGoPackage. Would you like to try that? There are plenty of examples in nixpkgs tree if you grep for buildGoModule.

File: pkgs/development/tools/github-commenter/default.nix (hunk 1 above)
Comment (onsails) [reply pkgs/development/tools/github-commenter/default.nix:23]: yeah, I've already done that in my other contributions. I'll upgrade here too.

**Review by prusnak — CHANGES_REQUESTED** [review]
the commit message should read `github-commenter: 0.5.0 -> 0.8.0` according to contributing policy

otherwise nice

File: pkgs/development/tools/github-commenter/default.nix [hunk 2]
```diff
@@ -1,16 +1,18 @@
...
     rev = version;
-    sha256 = "0y7yw7x8gqfbkqdfrwd9lffx3rrp62nz1aa86liy2dja97dacpij";
+    sha256 = "HgiCgyig+49g275G6zZ0kGTxt1TSfFK8kt+SOf4ei74=";
   };
 
+  vendorSha256 = "Gw+cR5sA5MGuclcvur8olmRtK04LDP5vKJ5k7yZO3B0=";
+
   goPackagePath = "github.com/cloudposse/${pname}";
 
   goDeps = ./deps.nix;
```
Comment (prusnak) [inline pkgs/development/tools/github-commenter/default.nix:18]: You can drop this line and remove the file `deps.nix` - these are not needed when using Go modules

**Review by prusnak — CHANGES_REQUESTED** [review]

File: pkgs/development/tools/github-commenter/default.nix (hunk 2 above)
Comment (onsails) [reply pkgs/development/tools/github-commenter/default.nix]: ah, indeed. done

File: pkgs/development/tools/github-commenter/default.nix [hunk 3]
```diff
@@ -1,25 +1,25 @@
...
     repo = pname;
     rev = version;
-    sha256 = "0y7yw7x8gqfbkqdfrwd9lffx3rrp62nz1aa86liy2dja97dacpij";
+    sha256 = "HgiCgyig+49g275G6zZ0kGTxt1TSfFK8kt+SOf4ei74=";
   };
 
-  goPackagePath = "github.com/cloudposse/${pname}";
+  vendorSha256 = "Gw+cR5sA5MGuclcvur8olmRtK04LDP5vKJ5k7yZO3B0=";
 
-  goDeps = ./deps.nix;
```
Comment (prusnak) [inline pkgs/development/tools/github-commenter/default.nix:16]: Please also delete the `deps.nix` file (`git rm`)

**Review by prusnak — CHANGES_REQUESTED** [review]

**Review by prusnak — APPROVED** [review]

**Pull request merged**